README.md

```diff
@@ -333,4 +333,353 @@ Other Style Guides
+    const [left, __, top] = processInput(input);
+
+    // good
+    function processInput(input) {
+      // then a miracle occurs
+      return { left, right, top, bottom };
+    }
+
+    // the caller selects only the data they need
+    const { left, right } = processInput(input);
+    ```
+
+
+**[⬆ back to top](#table-of-contents)**
+
+## Strings
+
+  - [6.1](#6.1) <a name='6.1'></a> Use single quotes `''` for strings.
+
+    ```javascript
+    // bad
+    const name = "Capt. Janeway";
+
+    // good
+    const name = 'Capt. Janeway';
+    ```
+
+  - [6.2](#6.2) <a name='6.2'></a> Strings longer than 100 characters should be written across multiple lines using string concatenation.
+  - [6.3](#6.3) <a name='6.3'></a> Note: If overused, long strings with concatenation could impact performance. [jsPerf](http://jsperf.com/ya-string-concat) & [Discussion](https://github.com/airbnb/javascript/issues/40).
+
+    ```javascript
+    // bad
+    const errorMessage = 'This is a super long error that was thrown because of Batman. When you stop to think about how Batman had anything to do with this, you would get nowhere fast.';
+
+    // bad
+    const errorMessage = 'This is a super long error that was thrown because \
+    of Batman. When you stop to think about how Batman had anything to do \
+    with this, you would get nowhere \
+    fast.';
+
+    // good
+    const errorMessage = 'This is a super long error that was thrown because ' +
+      'of Batman. When you stop to think about how Batman had anything to do ' +
+      'with this, you would get nowhere fast.';
+    ```
+
+  <a name="es6-template-literals"></a>
+  - [6.4](#6.4) <a name='6.4'></a> When programmatically building up strings, use template strings instead of concatenation.
+
+  > Why? Template strings give you a readable, concise syntax with proper newlines and string interpolation features.
+
+    ```javascript
+    // bad
+    function sayHi(name) {
+      return 'How are you, ' + name + '?';
+    }
+
+    // bad
+    function sayHi(name) {
+      return ['How are you, ', name, '?'].join();
+    }
+
+    // good
+    function sayHi(name) {
+      return `How are you, ${name}?`;
+    }
+    ```
+  - [6.5](#6.5) <a name='6.5'></a> Never use eval() on a string, it opens too many vulnerabilities.
+
+**[⬆ back to top](#table-of-contents)**
+
+## Functions
+
+  - [7.1](#7.1) <a name='7.1'></a> Use function declarations instead of function expressions.
+
+  > Why? Function declarations are named, so they're easier to identify in call stacks. Also, the whole body of a function declaration is hoisted, whereas only the reference of a function expression is hoisted. This rule makes it possible to always use [Arrow Functions](#arrow-functions) in place of function expressions.
+
+    ```javascript
+    // bad
+    const foo = function () {
+    };
+
+    // good
+    function foo() {
+    }
+    ```
+
+  - [7.2](#7.2) <a name='7.2'></a> Function expressions:
```

Comment:
I'm confused by this as well. I think though that this is simply about style, saying that instead of writing

```javascript
(function() {
    console.log('hello');
})();

you should use Arrow Functions.